### DISCUSSION TITLE
Not working in Qubes OS 4.0.4

### DISCUSSION THREAD

**Discussion by Arun1945** [OpenBangla/OpenBangla-Keyboard#240] [General]
Installed OpenBangla Keyboard in Fedora-33 TemplateVM . Can start the application in AppVM based on the template. Added OpenBangla Keyboard in GNOME desktop environment of AppVM as well as TemplateVM With the application running Super+Space keys does not change input source. Tested on Firefox browser and Text editor. Confused......

**Comment by mominul** [comment]
Have you tried restarting your Gnome shell?
Can you try running this in a terminal and try to type in the Text editor?
```
/usr/share/openbangla-keyboard/ibus-openbangla
```

Thanks!

**Reply by Arun1945** [reply]
টার্মিনালে এটা রান করালে হচ্ছে। কিন্তু টার্মিনাল এর কাজ বন্ধ হচ্ছে না। টার্মিনাল বন্ধ করে দিলে আর হচ্ছে না। কি করুম !!! 
If I run it in the AppVM terminal I can type according to the settings of OpenBangla-Keyboard. However the terminal does not drops down to the $ prompt. If I close the terminal by force, I loose the input source. I do not want the terminal to run all the time. Any solution ? Should I restart my AppVM ?

**Reply by Arun1945** [reply]
I have restarted the AppVM , now the command in the terminal throws up error and the Bangla input source lost.

**Reply by Arun1945** [reply]
Sometime the command works , sometimes not......

**Reply by Arun1945** [reply]
When the command do not work, I get the following errors: 

(process:1572): IBUS-CRITICAL **: 18:37:31.336: ibus_factory_new: assertion 'G_IS_DBUS_CONNECTION (connection)' failed

(process:1572): GLib-GObject-WARNING **: 18:37:31.337: invalid (NULL) pointer instance

(process:1572): GLib-GObject-CRITICAL **: 18:37:31.337: g_signal_connect_data: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed

(process:1572): IBUS-CRITICAL **: 18:37:31.337: ibus_bus_call_sync: assertion 'ibus_bus_is_connected (bus)' failed

(process:1572): IBUS-CRITICAL **: 18:37:31.337: ibus_bus_call_async: assertion 'ibus_bus_is_connected (bus)' failed

In addition to this I do not see the Avro-phonetic suggestion that appeared on the top left corner in Windows OS

**Comment by mominul** [comment]
I am suspecting that Qubes OS uses sandboxes to ensure security, and ibus cannot work because of the sandboxing. When you run the command in your terminal can you type Bengali?

**Reply by Arun1945** [reply]
Yes sometimes...but most of the times I get this error in the terminal that I have pasted in my previous reply. I have also tried to install the rpm package
 https://github.com/OpenBangla/OpenBangla-Keyboard/releases/download/2.0.0/OpenBangla-Keyboard_2.0.0-fedora33.rpm
The Software opens and I get signature cannot be verified.
Also , I have verified that ibus is not sandboxed in AppVMs

**Reply by mominul** [reply]
Your posted errors show that there is something wrong when establishing a connection between openbangla and ibus. Can you try seeking help from Qubes OS community, I think this is a distro-specific issue, and sandboxing or something related is in play here.

>> টার্মিনালে এটা রান করালে হচ্ছে। কিন্তু টার্মিনাল এর কাজ বন্ধ হচ্ছে না। টার্মিনাল বন্ধ করে দিলে আর হচ্ছে না। কি করুম !!!

You are creating an OpenBangla Keyboard instance by running that command, so exiting the terminal will also close the instance. You can explicitly detach the process by running the command following way:
```
/usr/share/openbangla-keyboard/ibus-openbangla &
```
I just wanted to know if the openbangla instance runs properly on your system by telling you to run the command to create an instance explicitly, so I could debug if there were some problems in the OpenBangla Keyboard. Seems like this was not a problem with OpenBangla Keyboard.

Thanks!

**Reply by Arun1945** [reply]
আবার কাজ করছে । অদ্ভুত !!! Its working again strange !!!

**Reply by mominul** [reply]
All is well that ends well! :grin:

**Reply by ahmubashshir** [reply]
try running `ibus-daemon -drx`

**Reply by Arun1945** [reply]
মমিনুল ভাই , আপনের থেইক্যা আহমেদ মুবাশির ভাই এর নলেজ বেশি। এই daemon ( আমি কই দানব) টা চালাইতেই সব একদম ঠিক হইয়া গেল। আহমেদ ভাই ও মমিনুল ভাই আপনাদের  দুজনকেই সালাম। এখন প্রশ্ন হচ্ছে আমি কি প্রতিবার AppVM চালাইবার সময় এই দানব টারে চালাইব নাকি একবারই চালাইতে হইবে ? কারণ AppVM নাকি TemplateVM এর ছাঁচেই কাজ করে , এবং বোধকরি কিছু সেটিং স ধইরা রাখতে পারে । নাকি এইটা আমি TemplateVM এই চালাইয়া দিব একবার ? এইটা আমারে জানাইবেন।

monimul I must say that ahmubashhir  has more knowledge in this matter. Running the daemon has solved all the issues. Salute to both of you !!! Should I run the code on each session of the AppVM as to my rudimentary knowledge AppVM can only store few settings but mostly run in the mould set by the TemplateVM. Please let me know.

**Reply by Arun1945** [reply]
I have another question. As my Fedora has GNOME desktop I had earlier added OpenBangla-Keyboard in the settings application. Why that was not enough ? I made the changes in the TemplateVM.

**Reply by ahmubashshir** [reply]
open `~/.xprofile` and put 
```sh
export QT_IM_MODULE=ibus
export GTK_IM_MODULE=ibus
export XMODIFIERS=@im=ibus
ibus-daemon -drx
```
open `~/.xinitrc` and put
```sh
source ~/.xprofile
```
then make them executable executable. Then logout and login.

**Reply by Arun1945** [reply]
[user@personal ~]$ xdg-open ~/.xprofile
gio: file:///home/user/.xprofile: Error when getting information for file “/home/user/.xprofile”: No such file or directory
[user@personal ~]$

**Reply by ahmubashshir** [reply]
use nano/vim/emacs/or your editor of choice.

**Reply by Arun1945** [reply]
Should I make them bash script executable ?

**Reply by ahmubashshir** [reply]
yes.

**Reply by rashidi77** [reply]
> Should I make them bash script executable ?

Did it work? Please let me know.
